### PR TITLE
2.5.0

### LINKED ISSUE #767: Case insensitive filtering

### LINKED ISSUE #768: 2.5.0 Updates
- [x] Feature flag updates

### PR DESCRIPTION
**Pull request by mhwelander**
* [x] gotchas for MySql and Postgres (cs collations for mysql and citext / c collation for postgres
* [x] Performance recommendations

### CODE REVIEW

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/050-filtering.mdx [hunk 1]
```diff
@@ -522,6 +524,35 @@ const result = await prisma.user.findMany({
...
 ```
 
+## Case-insensitive filtering
+
+Case-insensitive filtering is preview feature and [can only be used with PostgreSQL](case-sensitivity#options-for-case-insensitive-filtering). Add the highlighted feature flag to enable distinct querying capabilities:
+
+generator client {
+provider = "prisma-client-js"
+previewFeatures = ["insensitiveFilters"]
+}
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/050-filtering.mdx]:
````suggestion
```prisma
generator client {
  provider = "prisma-client-js"
|   previewFeatures = ["insensitiveFilters"]
}
```
````

Comment (janpio) [reply content/03-reference/01-tools-and-interfaces/02-prisma-client/050-filtering.mdx]: Not sure how to highlight a line.

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/050-filtering.mdx [hunk 2]
```diff
@@ -522,6 +524,35 @@ const result = await prisma.user.findMany({
 })
 ```
 
+## Case-insensitive filtering
+
+Case-insensitive filtering is preview feature and [can only be used with PostgreSQL](case-sensitivity#options-for-case-insensitive-filtering). Add the highlighted feature flag to enable distinct querying capabilities:
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/050-filtering.mdx]:
```suggestion
Case-insensitive filtering is preview feature and [can only be used with PostgreSQL](case-sensitivity#options-for-case-insensitive-filtering). Add the highlighted feature flag to enable case-insensitive filtering capabilities:
```

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/055-case-sensitivity.mdx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+<TopBlock>
+
+Case sensitivity affects **filtering** and **sorting** of data, and is determined by your [database collation](#database-collation-and-case-sensitivity). Sorting and filtering data yields different results depending on your settings:
+
+| Action          | Case sensitive                               | Case insensitive                             |
+|-----------------|----------------------------------------------|----------------------------------------------|
+| Sort ascending  | `Apple`, `Banana`, `apple pie`, `banana pie` | `Apple`, `apple pie`, `Banana`, `banana pie` |
+| Match `"apple"` | `apple`                                      | `Apple`, `apple`                             |
+
+[Prisma Client](.) respects your database collation. Options and recommendations for supporting **case-insensitive** filtering and sorting with Prisma Client depend on your [database provider](#options-for-case-insensitive-filtering).
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/055-case-sensitivity.mdx:17]: I do not think we support impacting the sorting yet: https://github.com/prisma/prisma-client-js/issues/841

Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/02-prisma-client/055-case-sensitivity.mdx:17]: We don't, but it sorting is still affected - there's a link to the sorting GH issue on the page. Shall I add it to the first mention of sorting maybe?

Comment (janpio) [reply content/03-reference/01-tools-and-interfaces/02-prisma-client/055-case-sensitivity.mdx:17]: No opinion. Only skimmed the text really - if it makes sense in the big context, this is perfectly fine.

**Comment by matthewmueller** [comment]
I think this also needs adjusting: https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/sorting#can-i-perform-case-insensitive-sorting to the new issue.

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/050-filtering.mdx [hunk 4]
```diff
@@ -522,6 +524,37 @@ const result = await prisma.user.findMany({
 })
 ```
 
+## Case-insensitive filtering
+
+Case-insensitive filtering is preview feature and [can only be used with PostgreSQL](case-sensitivity#options-for-case-insensitive-filtering). Add the highlighted feature flag to enable case-insensitive filtering capabilities:
```
Comment (Jolg42) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/050-filtering.mdx]:
```suggestion
Case-insensitive filtering is a preview feature and [can only be used with PostgreSQL](case-sensitivity#options-for-case-insensitive-filtering). Add the highlighted feature flag to enable case-insensitive filtering capabilities:
```
Maybe?

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/050-filtering.mdx [hunk 5]
```diff
@@ -522,6 +524,37 @@ const result = await prisma.user.findMany({
...
+
+```prisma
+generator client {
+  provider = "prisma-client-js"
+|   previewFeatures = ["insensitiveFilters"]
+}
+```
+
+Please help us make this feature production-ready by trying it out, [sharing your feedback](https://github.com/prisma/prisma/issues/3340) and [reporting bugs](https://www.prisma.io/docs/more/creating-bug-reports/)! The more and better feedback we receive, the earlier the feature can be released for production usage.
```
Comment (Jolg42) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/050-filtering.mdx]: In the release notes we tried to emphasize on you can post feedback if it works, maybe we could do the same?
 
>Please share your feedback on how this feature works for you. We are interested in both positive and negative feedback, so we know if this feature is already ready for production! (If encounter any problems, please open a new issue here).

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/055-case-sensitivity.mdx (hunk 3 above)
Comment (Jolg42) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/055-case-sensitivity.mdx:17]: `(.)` is that a missing link? or is it valid?

Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/02-prisma-client/055-case-sensitivity.mdx:17]: It's valid - `..` did not work, oddly, but `.` did

**Comment by Jolg42** [comment]
Any idea why the netlify preview doesn't work and says "Not Found"?
https://deploy-preview-772--prisma2.netlify.app/

File: content/03-reference/01-tools-and-interfaces/02-prisma-client/055-case-sensitivity.mdx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+### PostgreSQL provider
+
+PostgreSQL uses [deterministic collation](https://www.postgresql.org/docs/current/collation.html#COLLATION-NONDETERMINISTIC) by default, which means that filtering is **case-sensitive**.  To support case-insensitive filtering, use the `mode: 'insensitive'` property on a per-field basis.
+
+Case-insensitive querying is a preview feature. Add the highlighted feature flag to enable it:
+```prisma
+generator client {
+provider        = "prisma-client-js"
+previewFeatures = ["insensitiveFilters"]
+}
```
Comment (Jolg42) [inline content/03-reference/01-tools-and-interfaces/02-prisma-client/055-case-sensitivity.mdx]: It looks like the indentation could be improved here 😃

**Review by Jolg42 — APPROVED** [review]
Please make sure to update the 2.5.0 release notes and add a link to the insensivity page after merging 😃 
https://github.com/prisma/prisma/releases/tag/2.5.0

**Comment by janpio** [comment]
> Any idea why the netlify preview doesn't work and says "Not Found"?
> [deploy-preview-772--prisma2.netlify.app](https://deploy-preview-772--prisma2.netlify.app/)

We renamed the deployment this morning. Can now be found at: https://deploy-preview-772--prisma2-docs.netlify.app/

**Pull request merged**